BuildScript.UnitTests/MetadataValueCollectionTest.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// See the NOTICE file distributed with this work for additional information
+// regarding copyright ownership.  rubicon licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.  You may obtain a copy of the
+// License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+// License for the specific language governing permissions and limitations
+// under the License.
+//
+using System;
+using NUnit.Framework;
+using Remotion.BuildScript.BuildTasks;
+
+namespace Remotion.BuildScript.UnitTests
+{
+  [TestFixture]
+  public class MetadataValueCollectionTest
+  {
+    [Test]
+    [TestCase ("AAA", "AAA")]
+    [TestCase ("AAA", "aaa")]
+    [TestCase ("AaA", "aAa")]
+    public void Initialize_MultipleValues (string value1, string value2)
+    {
+      var rawCollection = new[] { value1, value2 };
+
+      Assert.That (
+          () => new MetadataValueCollection (rawCollection),
+          Throws.ArgumentException.With.Message.Contains ($"Duplicate values found in collection: '{value1}'.")
```

Comment:
should this test use more than one value in the duplicate test to show that the concat works?